src/js/actions/superselect.js

```diff
@@ -309,23 +310,31 @@ define(function (require, exports) {
                 });
             break;
         case kinds.SMARTOBJECT:
-            // For linked smart objects, this option shows the fix broken link dialog if the link is broken
-            var editOptions = {
-                interactionMode: descriptor.interactionMode.DISPLAY
-            };
-
-            _logSuperselect("edit_smart_object");
-            resultPromise = descriptor.play("placedLayerEditContents", {}, editOptions)
-                .bind(this)
-                .then(function () {
-                    // This updates the newly opened smart object document, although we should figure out a way
-                    // to check to see if it's being opened in Photoshop
-                    // Even if it's being opened in another app, the update call will not be visible to the user
-                    return this.transfer(documentActions.updateDocument);
-                }, function () {
-                    // We have an empty catch here, because PS throws cancel if user cancels on
-                    // Resolve Missing File dialog.
-                });
+            if (layer.isCloudLinkedSmartObject()) {
+                var elementReference = layer.getLibraryElementReference(),
+                    element = this.flux.stores.library.getElementByReference(elementReference);
+                    
+                resultPromise = this.transfer(libraryActions.openGraphicForEdit, element);
+            } else {
+                // For linked smart objects, this option shows the fix broken link dialog if the link is broken
+                
+                var editOptions = {
+                    interactionMode: descriptor.interactionMode.DISPLAY
+                };
+
+                _logSuperselect("edit_smart_object");
```

Comment:
This line should be called for all smart object types (or better yet, add a `"edit_cloud_object"` logging for CLSOs)